src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-overview-image/GeneOverviewImage.tsx

```diff
@@ -0,0 +1,125 @@
+import BasePairsRuler from 'src/content/app/entity-viewer/gene-view/components/base-pairs-ruler/BasePairsRuler';
+
+import { Gene } from 'src/content/app/entity-viewer/types/gene';
+import { Strand } from 'src/content/app/entity-viewer/types/strand';
+
+import styles from './GeneOverviewImage.scss';
+
+type GeneOverviewImageProps = {
+  gene: Gene;
+};
+
+const GeneOverviewImage = (props: GeneOverviewImageProps) => {
+  const { start: geneStart, end: geneEnd } = getFeatureCoordinates(props.gene); // FIXME: use gene length further on
+  const length = geneEnd - geneStart;
+
+  return (
+    <div className={styles.container}>
+      <GeneId {...props} />
+      <DirectionIndicator />
+      <GeneImage {...props} />
+      <StrandIndicator {...props} />
+      <NumberOfTranscripts {...props} />
+      <div className={styles.ruler}>
+        <BasePairsRuler length={length} width={695} standalone={true} />
+      </div>
+    </div>
+  );
+};
+
+export const GeneImage = (props: GeneOverviewImageProps) => {
+  const width = 695;
+  // const height = 18;
+  const { start: geneStart, end: geneEnd } = getFeatureCoordinates(props.gene);
+
+  // FIXME: use the "length" property of the gene when it is added to payload;
+  // (it will help with drawing genes of circular chromosomes)
+  const scale = scaleLinear()
+    .domain([geneStart, geneEnd])
+    .range([0, width]);
+
+  const renderedTranscripts = props.gene.transcripts.map(
+    (transcript, index) => {
+      const {
+        start: transcriptStart,
+        end: transcriptEnd
+      } = getFeatureCoordinates(transcript);
+      const startX = scale(transcriptStart);
+      const endX = scale(transcriptEnd);
+      const y = 10;
+      const width = Math.floor(endX - startX);
+      return (
+        <g key={index} transform={`translate(${startX} ${y})`}>
+          <UnsplicedTranscript
+            transcript={transcript}
+            width={width}
+            classNames={{
+              transcript: styles.transcript
+            }}
+          />
+        </g>
+      );
+    }
+  );
+
+  return (
+    <svg className={styles.containerSVG} width={width}>
+      {renderedTranscripts}
+    </svg>
+  );
+};
+
+const GeneId = (props: GeneOverviewImageProps) => (
+  <div className={styles.geneId}>{props.gene.id}</div>
+);
+
+const DirectionIndicator = () => {
+  return (
+    <div className={styles.directionIndicator}>
+      <span>5'</span>
+      <span>3'</span>
+    </div>
+  );
+};
+
+// FIXME translating response into display name (forward strand, reverse strandt) should be a shared function
+const StrandIndicator = (props: GeneOverviewImageProps) => {
+  const {
+    gene: {
+      slice: {
+        region: {
+          strand: { code: strandCode }
+        }
+      }
+    }
+  } = props;
+
+  let strandDisplayName;
+
+  if (strandCode === Strand.FORWARD) {
+    strandDisplayName = 'forward strand';
+  } else {
+    strandDisplayName = 'reverse strand';
+  }
```

Comment:
`const strandDisplayName = strandCode === Strand.FORWARD ?  'forward strand' : 'reverse strand'` ?

Reply:
✔️ 

I think it would be nice to move this logic to a shared helper; but I'm having troubles deciding on a name of the helper file. We have a pretty mixed bag of helpers in the `src/shared/helpers` folder:

![image](https://user-images.githubusercontent.com/6834224/75178299-c34e7600-572f-11ea-9d9e-ebdd0694ca91.png)

should we create something like "display-helpers" or "text-helpers" subfolder in that folder, and move `numberFormatter`, `regionFormatter`, and perhaps `textHelpers` in there? One other yet unwritten candidate is `pluralizationHelpers`, which will help us with choosing the right form of noun depending on its number (e.g. 1 transcript - 2 transcripts).

What do you think, @imransl , @msmanoj ?

Reply:
Creating subfolders is a good idea but we need to have some discussion about choosing the right names. Also, would it be helpful to add these to storybook?

Reply:
> Also, would it be helpful to add these to storybook?

I don't think so. They are not components. Tests that show how they behave should be enough.

Reply:
Fair enough. I thought it would be nice if there is any better way to improve the findability of these helper functions other than looking at the tests.